contracts/internals/controllable.sol

```diff
@@ -58,6 +48,16 @@ contract Controllable is ENSResolvable {
         return _controllerNode;
     }
 
+    /// @notice Initializes the controller contract object.
+    /// @param _controllerNode_ is the ENS node of the Controller.
+    /// @dev pass in bytes32(0) to use the default, production node labels for ENS
+    function _initializeControllable(bytes32 _controllerNode_) internal {
```

Comment:
I made changes in the controller v2 branch where I added default values to ENS variables.
IMO the _controllerNode should have a default value of controller.tokencard.eth and then we could remove the constructor and just have a separate method to setControllerNode() otherwise this intialize function is not safe because if you forget to invoke it in the constructor you will deploy a broken contract. Which isn't the case with a constructor because the compiler will tell you about it.

Reply:
I like that pattern, this is something that has caught us out before.

Reply:
Specifically for the controllable, there is a constant assigned during initialization. I left the `If` statement there even though I prefer a `require` one because it messes up with the unit tests and the tool suites (call me echidna). It was done in this way, so that wallet deployed can pass a 0 value and no problem would arise.